great_ape_safe/ape_api/uni_v3.py

```diff
@@ -399,3 +433,39 @@ def transfer_nft(self, token_id, new_owner):
 
         # assert new owner
         assert self.nonfungible_position_manager.ownerOf(token_id) == new_owner
+
+    def swap(self, path, mantissa, destination=None):
+        # https://docs.uniswap.org/protocol/guides/swaps/multihop-swaps
+        destination = self.safe.address if not destination else destination
+    
+        token_in, token_out = path[0], path[-1]
+
+        balance_token_out = token_out.balanceOf(self.safe)
+    
+        multihop_path = self._build_multihop_path(path)
+
+        path_encoded = b""
+        for item in multihop_path:
+            if web3.isAddress(item):
+                path_encoded += web3.toBytes(hexstr=item)
+            else:
+                path_encoded += int.to_bytes(item, 3, byteorder="big")
+
+        min_out = self.quoter.quoteExactInput(
+            path_encoded,
+            mantissa,
+        ).return_value * (1 - self.slippage)
```

Comment:
the quote function is actually (weirdly) not defined as a `view`, resulting in brownie performing an actual tx for it if we dont specify `.call`:
```suggestion
        min_out = self.quoter.quoteExactInput.call(
            path_encoded,
            mantissa,
        ) * (1 - self.slippage)
```

Reply:
good call